tap_mssql/__init__.py

```diff
@@ -101,6 +101,10 @@ def schema_for_column(c, config):
 
     if data_type == "bit":
         result.type = ["null", "boolean"]
+    
+    elif data_type in ["timestamp", "rowversion"]:
+        result.type = ["null", "string"]
+        result.format = "rowversion"
```

Comment:
Hi @mjsqu , do we know what affect the `result.format = "rowversion` will have on the target. Is this something like Target Snowflake is aware of?

Reply:
Would be good to run the pipeline into a target e.g. Target Snowflake.

Reply:
Thanks for reviewing this @mjsqu . This change is approved.

Reply:
## Integration Testing
Concerns have been raised around how target-snowflake might handle the updated code. To test this, we will generate some messages from the updated tap, based on a table with a timestamp column, and send these messages to target-snowflake:

- Create a new codespace on the tap-mssql merge request
- Update the inbuilt tests so that they produce SINGER_MESSAGES output to the terminal:
```
docker run -p 1433:1433 -e "MSSQL_SA_PASSWORD=testDatabase1" -e "ACCEPT_EULA=Y" mcr.microsoft.com/mssql/server:2019-latest
```
In another window:
```
poetry install
poetry run pytest -s -k TestTimestampIncrementalReplication
```
Copy singer SCHEMA, STATE and RECORD messages to files for scenario based testing:
### SCHEMA
```json
{
    "type": "SCHEMA",
    "stream": "dbo-incremental",
    "schema": {
        "properties": {
            "val": {
                "inclusion": "available",
                "minimum": -2147483648,
                "maximum": 2147483647,
                "type": [
                    "null",
                    "integer"
                ]
            },
            "updated": {
                "inclusion": "available",
                "format": "rowversion",
                "type": [
                    "null",
                    "string"
                ]
            }
        },
        "type": "object"
    },
    "key_properties": [],
    "bookmark_properties": [
        "updated"
    ]
}
```
### RECORD
```json
{
    "type": "RECORD",
    "stream": "dbo-incremental",
    "record": {
        "val": 1,
        "updated": "00000000000007d1"
    },
    "version": 1729723178861,
    "time_extracted": "2024-10-23T22:39:38.861597Z"
}
{
    "type": "RECORD",
    "stream": "dbo-incremental",
    "record": {
        "val": 2,
        "updated": "00000000000007d2"
    },
    "version": 1729723178861,
    "time_extracted": "2024-10-23T22:39:38.861597Z"
}
{
    "type": "RECORD",
    "stream": "dbo-incremental",
    "record": {
        "val": 3,
        "updated": "00000000000007d3"
    },
    "version": 1729723178861,
    "time_extracted": "2024-10-23T22:39:38.861597Z"
}
```
### STATE
```json
{
    "type": "STATE",
    "value": {
        "bookmarks": {
            "dbo-incremental": {
                "version": 1,
                "replication_key_value": "00000000000007d2",
                "replication_key": "updated"
            }
        },
        "currently_syncing": "dbo-incremental"
    }
}
```
- N.B. STATE message is not required as STATE messages are only consumed by taps to determine which RECORD messages to produce
- Run target-snowflake with the SCHEMA message and a limited number of RECORD messages as input

```
TARGET_SNOWFLAKE_DBNAME=dwh_dev_land_raw
TARGET_SNOWFLAKE_DEFAULT_TARGET_SCHEMA=dummy
```

The resulting table was produced and loaded with records without error